retrofit/src/main/java/retrofit2/Utils.java

```diff
@@ -82,7 +82,9 @@ static boolean equals(Type a, Type b) {
       if (!(b instanceof ParameterizedType)) return false;
       ParameterizedType pa = (ParameterizedType) a;
       ParameterizedType pb = (ParameterizedType) b;
-      return equal(pa.getOwnerType(), pb.getOwnerType())
+      Object ownerA = pa.getOwnerType();
+      Object ownerB = pb.getOwnerType();
+      return (ownerA == ownerB || (ownerA != null && ownerA.equals(ownerB)))
```

Comment:
could be `equals(pa.getOwnerType(), pb.getOwnerType())` (one character change).
also avoids Type.equals usage, though i highly doubt there would ever be real case of a broken owner type.